db/model/collector.js

```diff
@@ -130,17 +131,16 @@ module.exports = function collector(seq, dataTypes) {
 
       afterUpdate(inst /* , opts */) {
         if (inst.changed('status')) {
-          /* if status is changed to Running, then find and assign unassigned
-           generators */
           if (inst.status === collectorStatus.Running) {
+            /* if status is changed to Running, then find and assign unassigned
+             generators */
             return u.assignUnassignedGenerators();
-          }
-
-          if (inst.status === collectorStatus.Stopped ||
-            inst.status === collectorStatus.Paused) {
-            /* if status is changed to Stopped or Paused, then reassign the
-             generators which were assigned to this collector */
-            return inst.reassignGenerators();
+          } else if (inst.previous('status') === collectorStatus.Running) {
```

Comment:
Are we sure we want to unassign generators when a collector is paused? What's the purpose of "Paused" if it's functionally identical to "Stopped"?

Reply:
Paused == still sends heartbeats but doesn't generate samples (i.e. it can be resumed via api without needing access to the host running the collector)